torch/_export/graph_module.py

```diff
@@ -256,26 +279,32 @@ def make_export_graph_module(
 
     input_shape_constraints_by_src_name: Dict[str, List[Tuple[int, ConstraintExpr, ConstraintExpr]]] = {}
     input_name_to_example_inputs: Dict[str, Any] = {}
+    num_params_buffers = len(mutation_data.parameters) + len(mutation_data.buffers) if mutation_data is not None else 0
     if example_inputs is not None:
         input_tracker = 0
         for node in gm.graph.nodes:
             if node.op == "placeholder":
-                example_input = example_inputs[input_tracker]
-                if id(example_input) in input_shape_constraints_by_tensor_id:
-                    input_shape_constraints_by_src_name[node.name] = input_shape_constraints_by_tensor_id[id(example_input)]
-                input_name_to_example_inputs[node.name] = example_input
+                if input_tracker >= num_params_buffers:
+                    example_input = example_inputs[input_tracker - num_params_buffers]
+                    if id(example_input) in input_shape_constraints_by_tensor_id:
+                        input_shape_constraints_by_src_name[node.name] = input_shape_constraints_by_tensor_id[id(example_input)]
+                    input_name_to_example_inputs[node.name] = example_input
                 input_tracker += 1
 
     meta = ExportMetadata(
         in_spec=in_spec,
         out_spec=out_spec,
         update_spec=0,
-        mutation=mutation if mutation else [],
         input_shape_constraints=input_shape_constraints_by_src_name,
         inline_constraints=inline_constraints,
         input_name_to_example_inputs=input_name_to_example_inputs,
+        mutation_data=mutation_data,
     )
     attach_export_graph_metadata(gm, meta)
+    # TODO figure out where to save them
+
+    gm.parameters = params if params is not None else OrderedDict()
```

Comment:
QQ: what's the difference between params and buffers vs mutation_data.parameters , mutation_data.buffers?

Reply:
mutation_data.params and mutation_data.buffers are the names (str) of the params and buffers, not actual data